torchgeo/datasets/geo.py

```diff
@@ -342,7 +346,21 @@ class RasterDataset(GeoDataset):
     #: ``start`` and ``stop`` groups.
     date_format = "%Y%m%d"
 
-    #: True if dataset contains imagery, False if dataset contains mask
+    #: True if the dataset contains imagery or a digital surface, such as a Digital
+    #: Elevation Model, or another source of source data. False if the dataset
+    #: contains a mask,that is the target data, normally categorical data like land
+    #: cover classification.
+    #:
+    #: The value of ``is_image`` controls two things.
+    #:
+    #: The first is ``dtype``. See below for the impacts.
+    #:
+    #: The second is the name of the key used in the sample returned by
+    #: __getitem__. ``True`` uses "image"; ``False`` uses "mask". For consistency
+    #: purposes the same names as Kornia are used. When multiple datasets with different
+    #: keys are combined and the same key is used for multiple datasets, for example 2
+    #: "image" and 1 "mask", the channels will be stacked so that there's still a single
+    #: value for that key.
```

Comment:
I wonder if we can make this more succinct like the other attributes.

Reply:
Does this work?